### PR TITLE
introduced username

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: lib/Offer.js [hunk 1]
```diff
@@ -8,6 +10,11 @@ class Offer {
     }
   }
 
+  async getSellerName(marketAddress) {
+    var market = new MarketRepository().fromAddress(marketAddress);
+    this.sellerName = await market.getUserName(this.seller);
+    return this;
+  }
```
Comment (marekkirejczyk) [inline lib/Offer.js]: Should be perheps methiod of the market?

File: contracts/protocol/Market/Offer.sol [hunk 2]
```diff
@@ -46,7 +46,7 @@ contract Offer is Ownable {
         measurements = _measurements;
         requirements = _requirements;
         validator = _validator;
-        _market.pushOffer(this);
+        _market.pushOffer(this, seller);
```
Comment (marekkirejczyk) [inline contracts/protocol/Market/Offer.sol]: Perhaps we can inline seller and have:
`_market.pushOffer(this, msg.sender);`

File: contracts/protocol/Profile/Profile.sol [hunk 3]
```diff
@@ -6,6 +6,7 @@ import "../Utils/Ownable.sol";
 contract Profile is Ownable{
 
   EscrowedAgreement[] agreements;
+  bytes32 public name;
```
Comment (marekkirejczyk) [inline contracts/protocol/Profile/Profile.sol]: Why not string?

Comment (vanruch) [inline contracts/protocol/Profile/Profile.sol]: more resources; can't pass through external calls

**Pull request merged**